packages/serverpod/lib/src/server/server.dart

```diff
@@ -526,11 +527,8 @@ class Server {
       return ResultInvalidParams(e.message);
     } on NotAuthorizedException catch (e) {
       return e.authenticationFailedResult;
-    } on InvalidParametersException catch (e, stackTrace) {
```

Comment:
This change could in theory be a breaking change. Is the sessionLogId used / needed in this error scenario? Note that this code seems to be rather old, >4 ys, if I parsed the history correctly.

Reply:
TLDR; Change looks safe to me! 👍 

### Logging Id
Logically speaking I don't think logging the Session Id makes a lot of sense until the request has reached the endpoint. Before that the session hasn't really been "opened" even if the object is created.

The session id is also unique for each method call session created (the override is not used in this codepath): https://github.com/serverpod/serverpod/blob/a19e1660f236423349a65340a8b6a2fa9630a4a8/packages/serverpod/lib/src/server/session.dart#L145

So it won't help with debugging calls to the client.

In my opinion this is fine to remove.

### Logging in general
Logging changes level, we won't automatically print these to the server logs if a flag (configured through insights or server start) is not enabled. But I don't think this is a bad thing.

Now: https://github.com/serverpod/serverpod/blob/a19e1660f236423349a65340a8b6a2fa9630a4a8/packages/serverpod/lib/src/server/server.dart#L298-L302

Before: 
https://github.com/serverpod/serverpod/blob/a19e1660f236423349a65340a8b6a2fa9630a4a8/packages/serverpod/lib/src/server/server.dart#L320-L322

### Client representation
On the client side we will change the representation of these errors: https://github.com/serverpod/serverpod/blob/a19e1660f236423349a65340a8b6a2fa9630a4a8/packages/serverpod_client/lib/src/serverpod_client_shared_private.dart#L46-L53

But since this is an already known exception that had to be handled before I think this is completely fine.

Reply:
My only though is if we should return a 404 if the endpoint is not found? I'm not sure that makes a lot of sense since the endpoint dispatching depends on request body?

Reply:
404 when the path name (endpoint class name) is wrong would actually be more compliant with HTTP. Actual method name should be 400 as you say. I'll look into it!